client-js/app/scripts/app.ts

```diff
@@ -368,9 +400,23 @@ window.addEventListener('WebComponentsReady', () => {
       try {
         await attemptLogin(mgr, host, window.localStorage);
       } catch (e) {
-        await loginWithDialog(mgr, host);
+        if (topLevel) {
+          await loginWithDialog(mgr, host);
+        } else {
+          throw e;
+        }
       }
     }
+
+    var id = setInterval(() => {
```

Comment:
I had problems specifically on blacksand where the websocket connections would get closed after 30 seconds or so if there was nothing happening. I presume there is some http proxy on the corp network that causes this. These pings just go to the web server, they don't actually get sent along to labrad, so this just ensures that proxies don't think the websocket connection is idle and try to close it.